adserver/models.py

```diff
@@ -163,7 +163,13 @@ def keywords(self):
             return return_keywords
         return []
 
-    def daily_reports(self, start_date=None, end_date=None, campaign_type=None):
+    def daily_reports(
+        self,
+        start_date=None,
+        end_date=None,
+        campaign_type=None,
+        revenue_share_percentage=None,
```

Comment:
I don't think we want this here. This report is already specific to a single publisher. If we want to limit the publishers we are running this across, that should be done before this method is called.

Since this figure doesn't differ for a publisher, do we want this?

Reply:
It seems all the filtering is done in this function, so for the `All publishers` report, it seems like we need to pass it in?

Reply:
This function is per publisher. We need some filtering in the function because there could be both paid and house campaigns shown on the publisher. So if you want to filter by campaign type, that should be done on a per publisher basis because there are both paid and house impressions across the publisher. However, every impression for a publisher has the same revshare percentage so that could be done before it gets to here.

To make this concrete, [this is what I'm proposing](https://github.com/readthedocs/ethical-ad-server/compare/improve-filtering-options...davidfischer/revshare-filtering-alternative).

Reply:
Ah yea, thats fine -- I was just using the existing pattern.

Reply:
I turned it into a PR #171.